assets/src/models/createDetourMachine.ts

```diff
@@ -1,17 +1,142 @@
-import { setup } from "xstate"
+import { setup, assign, fromPromise } from "xstate"
+import { RouteId, RoutePattern, RoutePatternId } from "../schedule"
+import { fetchRoutePatterns } from "../api"
 
 export const createDetourMachine = setup({
   types: {} as {
-    events: { type: "detour.edit.done" } | { type: "detour.edit.resume" }
+    context: {
+      routeId?: RouteId
+      routePatternId?: RoutePatternId
+
+      routePatterns?: RoutePattern[]
+    }
```

Comment:
note: I'm not sure it's worth making this type "complex" like the `input` type?

I _think_ that doing more complex types here would make `assign` and other interactions with `context` harder, but hopefully it doesn't, but I'd like to leave that for a later PR.